project/src/test/java/com/google/sps/servlets/QueryServletTest.java

```diff
@@ -176,11 +183,41 @@ public void getRequest_placesAndUserLocationForwadedToScorer() throws Exception
     when(FETCHER.fetch(any(UserPreferences.class))).thenReturn(places);
     when(REQUEST.getParameter("location")).thenReturn("00.00000000,00.00000000");
 
-    servlet.doGet(REQUEST, RESPONSE);
+    servlet.doPost(REQUEST, RESPONSE);
 
     verify(SCORER).getScores(places, new LatLng(00, 00));
   }
 
+  @Test
+  // This test checks that storeUserPreferences is called with the expected parameters
+  // when the servlet gets a valid ID token
+  public void getRequest_validIdToken_userIdAndPreferencesForwardedForStoring() throws Exception {
+    initializeRequestParameters();
+    when(FETCHER.fetch(any(UserPreferences.class))).thenReturn(ImmutableList.of());
+    when(REQUEST.getParameter("idToken")).thenReturn("token");
+    when(USER_VERIFIER.getUserIdByToken("token")).thenReturn(Optional.of("userId"));
+
+    servlet.doPost(REQUEST, RESPONSE);
+
+    UserPreferences expectedUserPrefs = getValidUserPreferencesBuilder().build();
+    verify(DATA_ACCESSOR).storeUserPreferences("userId", expectedUserPrefs);
```

Comment:
The only thing that bothers me is that the two places that update the preferences are assumed to do the exact same thing, but they are independent and I think that's a problem. I'll explain: the function "initializeRequestParameters" initializes the preferences to be exactly the same as the preferences in "getValidUserPreferencesBuilder()", so this test works. but these two are independent, which means someone can come and change one of them a bit, for example change the rating to 5, and that wouldn't be a wrong thing to do, but it will cause this test to fail. 
I suggest an "ugly" solution, which in my opinion is more safe, which is to set the attributes inside the test, so that it's clear that changing something may effect the test results. Either do what the functions do (which is uglier), or make the functions receive input and send the same input to both, or something like that.

Reply:
Ok, I changed it as you suggested